generation/generate/generate.ts

```diff
@@ -89,7 +89,18 @@ export function generate(analysisResult: AnalysisResult): string {
         // hack to avoid yet another lifetime
         writer.writeLine("let bump_ref = unsafe { mem::transmute::<&Bump, &'a Bump>(&bump_borrow) };");
         writer.writeLine("let info_ref = unsafe { mem::transmute::<&ModuleInfo, &'a ModuleInfo<'a>>(&info) };");
-        writer.writeLine(`let ast_view = ${getViewForFunctionName("Module")}(info_ref, bump_ref);`);
+        writer.writeLine("let tokens = info_ref.tokens.map(|t| TokenContainer::new(t));");
+        writer.writeLine("let tokens_ref = tokens.as_ref().map(|t| unsafe { mem::transmute::<&TokenContainer, &'a TokenContainer<'a>>(t) });");
+        writer.write("let comments = info_ref.comments.map(|c| CommentContainer::new(");
+        writer.indent(() => {
+          writer.writeLine("c.leading,");
+          writer.writeLine("c.trailing,");
+          writer.writeLine(`tokens_ref.expect("Tokens must be provided when using comments."),`);
+          writer.writeLine(`info_ref.text_info.expect("Text info must be provided when using comments"),`);
+        });
+        writer.write("));").newLine();
+        writer.writeLine("let comments_ref = comments.as_ref().map(|t| unsafe { mem::transmute::<&CommentContainer, &'a CommentContainer<'a>>(t) });");
+        writer.writeLine(`let ast_view = ${getViewForFunctionName("Module")}(info_ref, tokens_ref, comments_ref, bump_ref);`);
```

Comment:
This is pretty bad. I will do a followup to remove all these transmutes and clean this up.